### ISSUE TITLE
New inspection: number of parameters in mysqli_stmt::bind_param()

### ISSUE THREAD

**Issue by BenMorel** [kalessil/phpinspectionsea#481]
It would be nice to add an inspection that checks that the number of bound parameters in [mysqli_stmt::bind_param()](http://php.net/manual/en/mysqli-stmt.bind-param.php) (and the procedural `mysqli_stmt_bind_param()` version, for that matter) matches the number of characters in the `$types` parameter:

```php
$statement = $mysqli->prepare(...);
$statement->bind_param('siii', $name, $age, $time); 
```

The above example would trigger a warning such as:

> Wrong number of bound parameters: expected 4, got 3

This would be similar to the existing inspection for `printf()`/`sprintf()`.

**Comment by kalessil** [comment]
I guess the `$types` validation makes sense as well.

**Comment by kalessil** [comment]
Thank you for reporting @BenMorel

**Comment by kalessil** [comment]
So, I put my thought a little bit regarding inspecting extensions-related APIs and decided to not open the Pandoras' box: we are not planning to support individual extensions APIs. For DB-related stuff, we'll stick to PDO only.
